docs/source/pika-comparison.rst

```diff
@@ -0,0 +1,288 @@
+
+.. _Pika: https://pika.readthedocs.io/en/stable/
+.. _RabbitMQ tutorials: https://www.rabbitmq.com/getstarted.html
+.. _work queues: https://www.rabbitmq.com/tutorials/tutorial-two-python.html
+.. _topics: https://www.rabbitmq.com/tutorials/tutorial-five-python.html
+.. _RPC: https://www.rabbitmq.com/tutorials/tutorial-six-python.html
+.. _aio-pika: https://aio-pika.readthedocs.io/en/latest/index.html
+
+Coming from Pika
+================
+
+KiwiPy comes natively with a RabbitMQ communicator (others can be added by extending the :py:class:`~kiwipy.Communicator` interface) and thus it may be useful to see how to achieve the same things in `Pika`_ (the standard library used in the `RabbitMQ tutorials`_) and kiwiPy.  This also shows some of the differences, particularly where kiwiPy is less verbose by setting sensible defaults for the user.
+
+
+Work Queues
+-----------
+
+Let's start with RabbitMQ's `work queues`_ example.
+
+Pika
+++++
+
+The code for sending a task::
+
+    import pika
+    import sys
+
+    connection = pika.BlockingConnection(
+        pika.ConnectionParameters(host='localhost'))
+    channel = connection.channel()
+
+    channel.queue_declare(queue='task_queue', durable=True)
+
+    message = ' '.join(sys.argv[1:]) or "Hello World!"
+    channel.basic_publish(
+        exchange='',
+        routing_key='task_queue',
+        body=message,
+        properties=pika.BasicProperties(
+            delivery_mode=2,  # make message persistent
+        ))
+    print(" [x] Sent %r" % message)
+    connection.close()
```

Comment:
Why not put these samples also in the `docs/source/examples` and `literalinclude` them like the `kiwipy` examples? It is also a bit confusing that the `kiwipy` snippets are in the folder `docs/source/examples/pika` :confused: 
I propose the following directory structure
```
examples/
    pika/
        rpc_server.py
        ....
    kiwipy/
        rpc_server.py
```
This way, it is easy to literalinclude the same snippet for each library next to each other and will it also be easy to extend the examples with another library if you'd want to in the future.

Reply:
Thanks, done.  I mainly didn't do this because the pika examples aren't expected to be runnable if the user just installs `kiwipy` but if they get as far as trying to execute those then they can figure out that they will need to install `pika`.